src/engine/strat_engine/backstore/blockdev.rs

```diff
@@ -129,6 +150,29 @@ impl StratBlockDev {
             Err(e) => return Err((e, bda)),
         };
 
+        let blksizes = match underlying_device {
+            UnderlyingDevice::Encrypted(_) => {
+                let metadata_path = underlying_device.metadata_path();
+                let crypt_blksizes = match OpenOptions::new()
+                    .read(true)
+                    .open(metadata_path)
+                    .map_err(StratisError::from)
+                    .and_then(|f| BlockSizes::read(&f))
+                {
+                    Ok(blksizes) => blksizes,
+                    Err(e) => return Err((e, bda)),
+                };
```

Comment:
We also want this information to be set properly when we have just initialized the device. Because there are really two places where we make a StratBlockDev, this seemed like the best place.

Reply:
Is it worth moving the physical block size in too then?

Reply:
I gave that a go. I think it might be a tad better. The thing is we need the sector sizes before we even make StratBlockDevs, because we need to have information about whether the sector sizes of the uninitialized devices allow them to be compatible with block devices already in the pool. So it just kind of shifts the code around, ultimately.

Reply:
Okay, this looks a little better to me. I'll approve it.